src/Native/Unix/configure.cmake

```diff
@@ -407,6 +407,18 @@ check_function_exists(
     getpeereid
     HAVE_GETPEEREID)
 
+check_function_exists(
+    getdomainname
+    HAVE_GETDOMAINNAME)
+
+check_function_exists(
+    uname
+    HAVE_UNAME)
+
+if(NOT HAVE_GETDOMAINNAME AND NOT HAVE_UNAME)
```

Comment:
That's a trick question - if I were to say yes, you could go and write your own Linux distro which as neither :).

That said, uname is part of POSIX: http://pubs.opengroup.org/onlinepubs/9699919799/functions/uname.html, so I'm guessing most Linux distributions will have it. 

But I don't think it can be guaranteed, so having `SystemNative_GetDomainName` return `E_NOTSUP` seems like an elegant way to let the .NET caller know that whatever platform this is running on, does not support that.